Updating optics and scatterers for Tutorial Review.

SkariZ opened the pull request:
#Add ISCAT and Darkfield classes to optics.py; update MieScatterer in scatterers.py

- Added the following classes to optics.py:
  - ISCAT: [Sets particle simulation to that of Iscat]
  - Darkfield: [Sets particle simulation to that of Darkfield]

- Updated the MieScatterer class in scatterers.py:
  - [Added computation on the scattering terms for Iscat and Darkfield]

SkariZ committed:
Merge pull request #1 from DeepTrackAI/develop

pr

SkariZ committed:
first_commit_review

BenjaminMidtvedt commented on deeptrack/optics.py:
```diff
@@ -723,6 +733,96 @@ def get(self, illuminated_volume, limits, fields, **kwargs):
+
+    Parameters
+    ----------
+    illumination : Feature
```
Add documentation for new parameters

BenjaminMidtvedt commented on deeptrack/optics.py:
```diff
@@ -723,6 +733,96 @@ def get(self, illuminated_volume, limits, fields, **kwargs):
+
+    Parameters
+    ----------
+    illumination : Feature
```
Add documentation for new parameter

BenjaminMidtvedt commented on deeptrack/scatterers.py:
```diff
@@ -540,6 +540,9 @@ def __init__(
         position_objective=(0, 0),
         return_fft=False,
         coherence_length=None,
+        illumination_angle=0,
+        is_iscat=False,
+        epsilon_iscat=1,
```
Add documentation. What does "is_iscat" mean? That you take the back scattered light? Try to name it more along the physical meaning than the specific modality

BenjaminMidtvedt reviewed (commented):
First set of review

SkariZ committed:
Fixing documentation for parameters and defining new parameters.

- Íscat is defined by setting the input/output polarization to circular and adding a phase_shift_correction term.
- Documentation for all variables

SkariZ committed:
Update optics.py
